src/subcommand.rs

```diff
@@ -205,6 +213,21 @@ Learn more at {}
                 .map(|_size_in_bytes| format!("{:x}", hasher.finalize()))?;
             println!("{}", hex_digest);
         }
+
+        Subcommand::Fetch => {
+            let file_arg = take_exactly_one_arg(args)?;
+            let dotslash_data = fs_ctx::read_to_string(file_arg)?;
+            let dotslash_cache = DotslashCache::new();
+            let (artifact_entry, artifact_location) =
+                locate_artifact(&dotslash_data, &dotslash_cache)?;
+            if !artifact_location.executable.exists() {
+                let provider_factory = DefaultProviderFactory {};
+                download_artifact(&artifact_entry, &artifact_location, &provider_factory)?;
+                eprintln!("Downloaded {}", artifact_location.executable.display());
+            } else {
+                eprintln!("Already exists: {}", artifact_location.executable.display());
+            }
```

Comment:
How do you feel about this? Stdout always just dumps the path (for easy parsing), and stderr tells you if it was downloaded or not. Or even even just doesn't tell you if it was downloaded.

```suggestion
            if !artifact_location.executable.exists() {
                let provider_factory = DefaultProviderFactory {};
                download_artifact(&artifact_entry, &artifact_location, &provider_factory)?;
                eprintln!("Downloaded artifact");
            } else {
                eprintln!("Artifact already exists");
            }
            println!("{}", artifact_location.executable.display());
```

Reply:
Great! I'll make these changes and land